### PR TITLE
Don't error when time limit exceeded on orderbook retrieval

### PR DESCRIPTION
**Pull request by nlordell**
The PR merged earlier today #647 causes false negative errors when CI starts right at the end of the solver time limit.

### Test Plan

CI, updated unit test.

### CODE REVIEW

**Review by e00E — APPROVED** [review]

File: driver/src/driver/stablex_driver.rs [hunk 1]
```diff
@@ -155,7 +155,8 @@ impl<'a> StableXDriver for StableXDriverImpl<'a> {
             Some(time_limit) => time_limit,
             None => {
                 self.metrics.auction_skipped(batch_to_solve);
-                return DriverResult::Skip(anyhow!("account retrieval exceeded time limit"));
+                warn!("account retrieval exceeded time limit");
```
Comment (fleupold) [inline driver/src/driver/stablex_driver.rs]: Nit: this should be orderbook retrieval not account retrieval

File: driver/src/driver/stablex_driver.rs [hunk 2]
```diff
@@ -155,7 +155,8 @@ impl<'a> StableXDriver for StableXDriverImpl<'a> {
             Some(time_limit) => time_limit,
             None => {
                 self.metrics.auction_skipped(batch_to_solve);
```
Comment (fleupold) [inline driver/src/driver/stablex_driver.rs]: I might have missed this in an earlier PR but this will cause us to log a successful processing (which is not true as we didn't attempt to solve this batch). Maybe we should rename the method to something like `processed_but_not_submitted` to make this clearer.

File: driver/src/driver/stablex_driver.rs [hunk 3]
```diff
@@ -155,7 +155,8 @@ impl<'a> StableXDriver for StableXDriverImpl<'a> {
...
-                return DriverResult::Skip(anyhow!("account retrieval exceeded time limit"));
```
Comment (fleupold) [inline driver/src/driver/stablex_driver.rs]: Maybe `Skip` here can also be misleading. Maybe the DriverResult should be ErrorRetryable, ErrorNonRetryable instead to make clear that Skip here is not benign

**Review by fleupold — APPROVED** [review]
LGTM, but noticed two things unrelated to this PR, which I think we should address.

File: driver/src/driver/stablex_driver.rs (hunk 3 above)
Comment (nlordell) [reply driver/src/driver/stablex_driver.rs]: I'll do this in a separate PR.

File: driver/src/driver/stablex_driver.rs (hunk 2 above)
Comment (nlordell) [reply driver/src/driver/stablex_driver.rs]: I noticed this as well but wanted to tackle this in a separate PR as I think it requires a bit of though. Right now we have some inconsistencies with how we handle timeouts:
- If the scheduler runs at the end of a batch past the `solver_time_limit`, it will silently (as in no metrics recorded) wait until the next batch
- If the scheduler runs at the end of a batch near the `solver_time_limit` then a couple things can happen:
  1. The orderbook retrieval is in time, but the solver is started with a very low time limit, the solver will probably not find a solution and it will increment the `auction_skipped` metric
  2. The orderbook retrieval puts us over the time limit, so we skip the solving and it will increment a `auction_skipped` metric.

To m,e both cases when the driver attempts to solve at the end of the batch, whether it starts the solver with a very low time limit (for arguments sake, lets say 0.1s) or skips it is about the same. 

A problem with this approach is that we don't really distinguish the case where because of some unforeseen issue, the orderbook retrieval takes, for argument's sake, 3 minutes and causes is to not run the solver. This should be an error but is currently equivalent to the second case mentioned above.

Also, the way we submit metrics is getting a little hard to follow for me, with many branches. I wanted to take a little time to clean this up and maybe address your concerns in the process.

Comment (nlordell) [reply driver/src/driver/stablex_driver.rs]: Let me know if you are OK with me addressing this in a separate PR.

Comment (fleupold) [reply driver/src/driver/stablex_driver.rs]: Yes, let's do that in a separate PR. Maybe we should re-introduce a minimum time in which we don't start the solver? We are also running into somewhat benign failures on rinkeby today. 

When we start fetching the orderbook with 3s remaining we expectedly run over time and still log an error.

File: driver/src/driver/stablex_driver.rs [hunk 4]
```diff
@@ -155,7 +155,8 @@ impl<'a> StableXDriver for StableXDriverImpl<'a> {
...
+                warn!("orderbook retrieval exceeded time limit");
```
Comment (bh2smith) [inline driver/src/driver/stablex_driver.rs]: It seems that the "account retrieval exceeded time limit" has become "orderbook retrieval exceeded time limit". Was this intended? Are these the same thing?

Comment (nlordell) [reply driver/src/driver/stablex_driver.rs]: It was intended. We are calling "`orderbook` retrieval" the process of retrieving the accounts (users + balances) and orders. The orderbook retrieval is what is taking too long and causing this warning message.
